android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -215,6 +222,11 @@ void destroy() {
         if (mCookieListOptInPageAndroidHandler != null) {
             mCookieListOptInPageAndroidHandler.close();
         }
+
+        if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_PLAYLIST)
```

Comment:
I don't think we need an extra check on feature enabled or not here. It brings us an extra access to native part plus it's unnecessary, just check on `mPlaylistPageHandler != null` works